packages/@lwc/engine-dom/src/custom-elements/create-custom-element-vanilla.ts

```diff
@@ -17,44 +17,57 @@ let elementBeingUpgradedByLWC = false;
 // `customElements.define('x-foo')`, then you don't have access to the upgradeCallback, so it's a dummy custom element.
 // This class should be created once per tag name.
 const createUpgradableConstructor = (
-    connectedCallback: LifecycleCallback,
-    disconnectedCallback: LifecycleCallback
+    connectedCallback?: LifecycleCallback,
+    disconnectedCallback?: LifecycleCallback
 ) => {
+    const hasConnectedCallback = !isUndefined(connectedCallback);
+    const hasDisconnectedCallback = !isUndefined(disconnectedCallback);
+
     // TODO [#2972]: this class should expose observedAttributes as necessary
-    return class UpgradableConstructor extends HTMLElement {
+    class UpgradableConstructor extends HTMLElement {
         constructor(upgradeCallback: LifecycleCallback) {
             super();
             // If the element is not created using lwc.createElement(), e.g. `document.createElement('x-foo')`,
             // then elementBeingUpgraded will be false
             if (elementBeingUpgradedByLWC) {
                 upgradeCallback(this);
-            } else {
+            } else if (hasConnectedCallback && hasDisconnectedCallback) {
```

Comment:
It is not clear why this has to be checked to decide if an element was upgraded outside of LWC.

Reply:
I'll add an extra comment. We don't need to do any work here if there is no connected or disconnected callback.

Reply:
Added a comment, also switched the `&&` to a `||` because that's more correct, even though the two booleans will always be the same.